webapi/src/main/scala/org/knora/webapi/responders/v2/SearchResponderV2.scala

```diff
@@ -824,20 +824,23 @@ class SearchResponderV2(responderData: ResponderData) extends ResponderWithStand
 
             // _ = println(searchResourceByLabelSparql)
 
-            searchResourceByLabelResponse: SparqlConstructResponse <- (storeManager ? SparqlConstructRequest(searchResourceByLabelSparql)).mapTo[SparqlConstructResponse]
+            searchResourceByLabelResponse: SparqlExtendedConstructResponse <- (storeManager ? SparqlExtendedConstructRequest(searchResourceByLabelSparql)).mapTo[SparqlExtendedConstructResponse]
 
             // collect the IRIs of main resources returned
             mainResourceIris: Set[IRI] = searchResourceByLabelResponse.statements.foldLeft(Set.empty[IRI]) {
-                case (acc: Set[IRI], (subjIri: IRI, assertions: Seq[(IRI, String)])) =>
-                    //statement.pred == OntologyConstants.KnoraBase.IsMainResource && statement.obj.toBoolean
-
+                case (acc: Set[IRI], (subject: SubjectV2, assertions: Map[SmartIri, Seq[LiteralV2]])) =>
                     // check if the assertions represent a main resource and include its IRI if so
-                    val subjectIsMainResource: Boolean = assertions.contains((OntologyConstants.Rdf.Type, OntologyConstants.KnoraBase.Resource)) && assertions.exists {
-                        case (pred, obj) =>
-                            pred == OntologyConstants.KnoraBase.IsMainResource && obj.toBoolean
```

Comment:
So with the type provided by the triplestore, no additional type conversion has to be performed, right?

Reply:
I’m on the train but if you have IntelliJ add a type annotation for `obj`, it‘s a `LiteralV2`, right? Look at the implementation of its `toBoolean` method. If the `LiteralV2` is a `BooleanLiteralV2`, the conversion has already been done in `HttpTriplestoreConnector` when the Turtle was parsed.

Reply:
Sorry, got confused there, ignore my reply above. But yes, the conversions are done in `HttpTriplestoreConnector`, using the types provided by the triplestore.

Reply:
ok, no problem